webapp/javascript/ui/LoadingOverlay.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useEffect, useState } from 'react';
+import LoadingSpinner from '@webapp/ui/LoadingSpinner';
+import cx from 'classnames';
+import styles from './LoadingOverlay.module.css';
+
+/**
+ * LoadingOverlay when 'active' will cover the entire parent's width/height with
+ * an overlay and a loading spinner
+ */
+export function LoadingOverlay({
+  active = true,
+  spinnerPosition = 'center',
+  children,
+  delay = 250,
+}: {
+  /** where to position the spinner. use baseline when the component's vertical center is outside the viewport */
+  spinnerPosition?: 'center' | 'baseline';
+  active: boolean;
+  children?: React.ReactNode;
+  /** delay in ms before showing the overlay. this evicts a flick */
+  delay?: number;
+}) {
+  const [isVisible, setVisible] = useState(false);
+
+  // Wait for `delay` ms before showing the overlay
+  // So that it feels snappy when server is fast
+  // https://www.nngroup.com/articles/progress-indicators/
+  useEffect(() => {
```

Comment:
delay is not required, so maybe it makes sense to not to use setTimeout at all if you don't pass `delay` prop? actually I'm not a fan of hardcoding any kind of static value of loading time :)

Reply:
Actually the default is 250ms. This is intentional and a design choice. For fast operations (<250ms we don't want to show any loading indication).

From Nielson Norman group:
![image](https://user-images.githubusercontent.com/6951209/195654197-a8e8584a-3c2d-4bea-ab40-61fe30e756f2.png)

Same principle, but I found 1 second too broad so changed for `250ms`.